### ISSUE TITLE
pick an SMTP server per campaign

### ISSUE THREAD

**Issue by jucajuca** [knadh/listmonk#1703]
**Is your feature request related to a problem? Please describe.**
A user can have 2 campaigns and 2 SMTP servers. The user would like to set CAMP1 to use SMTP1 and CAMP2 to use SMTP2.

**Describe the solution you'd like**
The user should be able to pick an SMTP server per campaign. 

Related to issue: https://github.com/knadh/listmonk/issues/259
@charlesmudy

**Comment by charlesmudy** [comment]
Still waiting for this and thought would come with v3

It's quite logical to have the option to choose an SMTP per campaign considering there are multiple SMTP servers already listed.

@knadh

**Comment by ondrejvrabel** [comment]
This would be useful for us too.

**Comment by iloveitaly** [comment]
Yeah, this is important for me as well. Using this across a couple of different gmail accounts and want to use the gmail SMTP option, one for each account (even better would be to assign SMTP per list).

**Comment by Satishglk** [comment]
Looking forward to this feature !

**Comment by tudor-cel-stan** [comment]
I personally find this feature to be a priority. Barely maintained tools like Sendy are still being used because of their versatility when it comes to sending from multiple addresses, domains and email servers, which makes client management a breeze. Having a separate install for each client is not practical at all, while reconfiguring the app settings before sending a campaign is even less practical.

Please take this into consideration. The templating features that you integrated in the app are insanely powerful. Maybe also take into consideration providing a managed Listmonk service on your servers since that could be a good revenue stream for your amazing product.

**Comment by vincentmarkcelino** [comment]
is this available on the latest version? we badly need this feature... we use different email for transactional and campaign. T_T

**Comment by charlesmudy** [comment]
I mean, @knadh has done incredible work with Listmonk so kudos for that but I honestly do not understand this is hard to implement when people have repeatedly requested for it.

We simply just added a new ID for smtp

<img width="997" alt="Screenshot 2024-11-30 at 14 20 10" src="https://github.com/user-attachments/assets/453b5b46-017b-4d66-ac72-e9e3a8abf7b4">

Then did this at line 102 inside email.go file

`
// Choose SMTP
for _, srvs := range e.servers {
	s := srvs
	if s.ID == m.Headers.Get("REGION") { // compare the ID with the campaign smtp or whatever your attribute is
		srv = s
		break
	}
}
`

**Comment by filipesmedeiros** [comment]
@charlesmudy why not create a PR?

**Comment by knadh** [comment]
This is now merged in #2290. Thanks to @lcd1232 for initiating the PR.

This patch adds a new optional `name` field to SMTP server config on the UI. When a name is given to an SMTP server, it's initialized as a standalone messenger which shows up as a sub-group item under the main "email" messenger on the campaign page.

**Comment by castellinosoftware** [comment]
> This is now merged in [#2290](https://github.com/knadh/listmonk/pull/2290). Thanks to [@lcd1232](https://github.com/lcd1232) for initiating the PR.
> 
> This patch adds a new optional `name` field to SMTP server config on the UI. When a name is given to an SMTP server, it's initialized as a standalone messenger which shows up as a sub-group item under the main "email" messenger on the campaign page.

Is it possible to set the SMTP server in a campaign created via API?

**Comment by knadh** [comment]
Yes. Campaign POST/PUT API takes the `messenger` field. Will add this to the docs when the release is finalised.

**Comment by sroquesleo** [comment]
is it possible to use 2 specific servers by naming them equally?

**Comment by lcd1232** [comment]
@sroquesleo what's your use case by naming them equally?

**Comment by sroquesleo** [comment]
> Settings Screen: ![image](https://github.com/knadh/listmonk/assets/3868003/86fc1775-a7bf-4cd3-ba88-3d61cfd3916c)

@lcd1232 like  the suggestion above where emails could be sent distributed between 2 specific senders. Also I had same problem as #2547 by testing a new server because my default one does not have permission on the other
